File: src/main/java/guri/kafka/blockchainchat/domain/Transaction.java
package guri.kafka.blockchainchat.domain;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "transaction")
public class Transaction {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private Long buyOrderId;
    private Long sellOrderId;
    private int amount;
    private int price;
}

File: src/main/kotlin/guri/kafka/blockchainchat/transaction/Transaction.kt
package guri.kafka.blockchainchat.transaction

import org.springframework.data.annotation.Id
import org.springframework.data.relational.core.mapping.Table

@Table
class Transaction(
    @Id
    var id: Long? = null,
    var buyOrderId: Long,
    var sellOrderId: Long,
    var amount: Int,
    var price: Int
) {
}